core/deployment/src/main/java/io/quarkus/runner/RuntimeClassLoader.java

```diff
@@ -486,4 +494,19 @@ public InputStream getInputStream() throws IOException {
         }
         return null;
     }
+
+    private ProtectionDomain createDefaultProtectionDomain(Path applicationClasspath) {
+        URL url = null;
+        if (applicationClasspath != null) {
+            try {
+                URI uri = new URI("file://" + applicationClasspath);
```

Comment:
Because of possible encoding issues, the `URI(scheme,host,path,fragment)` constructor should be used instead.  Just give `null` for `host` and `fragment`.